pkg/apis/deviceplugin/v1/qatdeviceplugin_webhook.go

```diff
@@ -91,5 +91,23 @@ func (r *QatDevicePlugin) validatePlugin() error {
 		}
 	}
 
+	if len(r.Spec.ProvisioningConfig) > 0 {
+		if len(r.Spec.InitImage) == 0 {
+			return errors.Errorf("ProvisioningConfig is set with no InitImage")
+		}
+		// if 4xxxvf is not enabled
+		contains := false
+
+		for _, kernelVfDriver := range r.Spec.KernelVfDrivers {
+			if kernelVfDriver != "4xxxvf" {
```

Comment:
this logic does not look right. for example with the default CR we have it triggers an error:

```
  kernelVfDrivers:
    - c6xxvf
    - 4xxxvf
```

perhaps follow what we have in https://github.com/intel/intel-device-plugins-for-kubernetes/blob/92b1371b92ff4bc9a431404eb67584ced8981a76/cmd/qat_plugin/dpdkdrv/dpdkdrv.go#L401:L407 for looking for values in a set.

Reply:
Ah, idk why I put != it should be ==
But, I am just thinking about how to change as you mentioned.
The example you said is map[string]struct{} and KernelVfDrivers are just []KernelVfDriver.

Reply:
You could loop all the kernelvfdrivers as now but check from a map for the ones that can be configured